### ISSUE TITLE
Usage difficulties

### ISSUE THREAD

**Issue by SomeUserName1** [Ericsson/CodeCompass#280]
Hi, i am currently trying to parse a framework but am experiencing difficulties:
```
keepalive /usr/local/bin/CodeCompass_parser -w ../.opencog/ -n OpenCog -i cogutil/ -i atomspace/ -i as-moses/ -i opencog/ -d "psql:host=localhost;port=5432;user=compass;password=12345678;database=mydatabase" -j4 
[ERROR] Database and working directory existence are inconsistent. Use -f for reparsing!
```
I set up the psql db like
```
sudo su postgres
psql

CREATE DATABASE mydatabase
  WITH ENCODING 'SQL_ASCII' LC_CTYPE 'C' LC_COLLATE 'C'
  TEMPLATE template0;

CREATE USER compass WITH SUPERUSER LOGIN PASSWORD 'mypassword';
```
As workspace folder i just created a new "normal" folder like
```
mkdir .opencog
```
Also triewd the -f option, but that also didnt help... 
What am i doing wrong?

Thanks in advance & best,
Fabian

**Comment by whisperity** [comment]
If you are logging in to the database with a user which has the `SUPERUSER` permission, do not create the database, and neither the "workspace" folder. CodeCompass will create both when the parsing starts.

**Comment by SomeUserName1** [comment]
If i dont specify a directory, it complains about a missing argument.:
```
[ERROR] Error in command line arguments: the option '--workspace' is required but missing
```
If i specify a none existing one, the database and workspace inconsistency error is back...

So what would be the changes i have to make? 
```
keepalive /usr/local/bin/CodeCompass_parser -w <???> -n OpenCog -i cogutil/ -i atomspace/ -i as-moses/ -i opencog/ -d "psql:host=localhost;port=5432;user=compass;password=12345678;" -j4
```

Probably it's better to stick to the creating by hand method, isnt it?

**Comment by whisperity** [comment]
The existence of data folder and database has been changed by a recent patch. @mcserep might be able to help on this.

**Comment by mcserep** [comment]
Indeed, because of recent changes in the code base in order to support incremental parsing, the existence of the workspace directory and the database must be consistent, so both of them must exist or neither of them should. In case of an initial parsing I advise you that none of them should exist, and CodeCompass will create both the workspace directory and the database.
(The documentation might be outdated in this aspect, a review might be necessary.)

**Comment by SomeUserName1** [comment]
@mcserep i dont want to be annoying but it d be very helpfull to know what args to pass exactly...
If i specify a non-existent dir and no database it's still throwing the workspace and db inconsistency error
